vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/GenericWaitStepsTests.java

```diff
@@ -75,4 +81,26 @@ void testWaitTillElementDisappears()
         waitSteps.waitForElementDisappearance(locator);
         verify(waitActions).wait(webElement, condition);
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void testElementByLocatorExistsForDuration()
+    {
+        Locator locator = new Locator(TestLocatorType.SEARCH, VALUE);
+        IExpectedSearchContextCondition<List<WebElement>> presenceOfAllElements = mock(
+                IExpectedSearchContextCondition.class);
+        IExpectedSearchContextCondition<Boolean> not = mock(IExpectedSearchContextCondition.class);
+        WaitResult<Boolean> waitResult = new WaitResult<>();
+        waitResult.setWaitPassed(true);
+
+        when(uiContext.getSearchContext()).thenReturn(webElement);
+        when(expectedSearchActionsConditions.presenceOfAllElementsLocatedBy(locator)).thenReturn(presenceOfAllElements);
+        when(expectedSearchActionsConditions.not(presenceOfAllElements)).thenReturn(not);
+        when(waitActions.wait(webElement, Duration.ZERO, not, false)).thenReturn(waitResult);
+
+        waitSteps.doesElementByLocatorExistsForDuration(locator, Duration.ZERO);
+
+        verify(softAssert).assertFalse("Element located by locator ' Search: 'value'; Visibility: VISIBLE;' has existed"
```

Comment:
I believe we should revise `org.vividus.ui.action.search.Locator#toString` as first